### PR TITLE
Add a status text for autorun

### PR DESCRIPTION
**Pull request by Amaz1**
I don't know if this is the correct way of implementing this, but I find that it is useful to know if you've enabled/disabled autorun.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -2851,6 +2850,16 @@ void Game::toggleCinematic(float *statustext_time)
...
 
+// Add WoW-style autorun by toggling continuous forward.
+void Game::toggleAutorun(float *statustext_time)
+{
+	static const wchar_t *msg[] = { L"autorun disabled", L"autorun enabled" };
+	bool autorun = !g_settings->getBool("continuous_forward");
+	g_settings->set("continuous_forward", bool_to_cstr(autorun));
+
+	*statustext_time = 0;
+	statustext = msg[autorun];
```
Comment (PilzAdam) [inline src/game.cpp]: Is this guaranteed to be 0 or 1? I would do it like this:
```C++
statustext = msg[autorun ? 1 : 0];
```

Comment (Zeno-) [inline src/game.cpp]: The C++ Standard says it is guaranteed be 0 or 1 (1 for true). That said, your suggestion is probably easier to read (in this specific case) and the compiler will (probably) optimise it anyway.

**Comment by est31** [comment]
Looks good :+1:

File: src/game.cpp (hunk 1 above)
Comment (PilzAdam) [inline src/game.cpp]: `== true` just looks like you don't understand what booleans are.

Comment (rubenwardy) [inline src/game.cpp]: or name it like `is_autorun` or `autorun_enabled` so it's more obvious that it's a flag.

Comment (Amaz1) [inline src/game.cpp]: Thanks. I've gone with @PilzAdam's method, and renamed it to autorun_enabled.

**Comment by paramat** [comment]
:-1: It seems obvious when autorun is enabled or disabled from player motion.

**Comment by rubenwardy** [comment]
:+1: It's consistent with all the other status toggles.

**Comment by Amaz1** [comment]
Even if it is obvious from player motion, I think that it should be added at least for consistency. Other effects that change movement have a status text, even if they are obvious (e.g. cinematic, or the HUD one, which is really obvious if it is enabled/disabled), so why should autorun be any different?

**Comment by est31** [comment]
Yes, it is obvious as paramat says, but if we went after obviousness, we wouldn't display that fly mode is disabled, because you'll start falling immediately if in the air, either. Consistency is important IMO.

**Comment by 0-afflatus** [comment]
:+1: for consistency.

**Comment by paramat** [comment]
Good points, i remove my -1.

**Comment by est31** [comment]
e664abeb887fe74ff1966e93ed5201c6e19a5c31

**Pull request closed without merging**